ARCs/arc-0004.md

```diff
@@ -581,18 +581,13 @@ For any ABI value `x`, we recursively define `enc(x)` to be as follows:
 * If `x` is a boolean value `bool`:
     * `enc(x)` is a single byte whose **most significant bit** is either 1 or 0, if `x` is true or false respectively. All other bits are 0. Note: this means that a value of true will be encoded as `0x80` (`10000000` in binary) and a value of false will be encoded as `0x00`. This is in contrast to most other encoding schemes, where a value of true is encoded as `0x01`.
 
-Since `string` and `address` are aliases for `byte[]` and `byte[32]`
-respectively (and `byte` is an alias for `uint8`), the rules for
-encoding these types are covered above.
+Other aliased types' encodings are already covered:
+- `string` and `address` are aliases for `byte[]` and `byte[32]` respectively
+- `byte` is an alias for `uint8`
+- each of the reference types is an alias for `uint8`
 
 ### Reference Types
 
-Three special types are supported _only_ as the type of an
-argument. They **MUST NOT** be embedded in arrays or tuples, as 
-their encoding is unusual. (They **MAY** implicitly be part of
-a tuple if they are the type of the 15-th argument or an argument
-further away.)
```

Comment:
Could you still mention that reference types are valid only in arguments (not in return values), and could you explicitly state that embedding these types in tuples or arrays is allowed?

Reply:
I misinterpreted the part that said that it can only be a type of the argument, meaning it cannot be a return type. Will fix.

Reply:
I'm on board after noting that somewhere that they cannot be return types.